csrc/ir/nodes.cpp

```diff
@@ -4540,8 +4541,12 @@ std::vector<PolymorphicValue> MatmulOp::evaluate(
   const auto& [sizes, strides] = inferShapeOfOutput(out(), ee);
   auto meta_out = at::detail::empty_strided_meta(sizes, strides, a.dtype());
   if (meta_out.is_contiguous()) {
-    return {matmul_out};
+    auto sharded_axis = getShardedLogicalAxis(out(), ParallelType::DIDx);
+    return {matmul_out.unsqueeze(sharded_axis)};
```

Comment:
This doesn't look right. Suppose

tv_a: a rowwise sharded [m, k]
tv_b: a replicated [k, n]
tv_out: therefore, a rowwise sharded [m, n]

Why should `tensor_out` be [1, m, n] instead?

Reply:
It's a missing commit, I just realized, fixing this.